search.go

```diff
@@ -59,6 +59,8 @@ func popSearchField(fields []interface{}) (interface{}, []interface{}, error) {
 // SearchCriteria is a search criteria. A message matches the criteria if and
 // only if it matches each one of its fields.
 type SearchCriteria struct {
+	All bool // Search for ALL messages in mailbox
```

Comment:
I don't think it's necessary to have a `All` field, since this is the default if no sequence number set or UID set is specified.

Reply:
I run UidSearch() with an empty SearchCriteria object. See details in the issue descr. Which default do you mean? I haven't found such a default description in RFC 3501.

Reply:
I think we should automatically append `ALL` to the list of fields if empty.

Reply:
Alright, makes sense. I'll update.